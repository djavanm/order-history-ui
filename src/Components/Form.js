import React, { Componenet } from 'react';
import './Form.css';

export default class Form extends Component {
  constructor() {
    super()

    this.state = {
      url: '',
      name: '',
      description: '',
      price: '',
    }

    handleInput = (e) => {
      e.preventDefault()
      this.setState({
        [e.target.name]: [e.target.value]
      })
    }
  }

  render() {
    const {url, name, description, price} = this.state;
    <form>
      <input
      className="form-input"
      type="text"
      placeholder="Image URL"
      onChange={e => this.handleInput(e)}
      name="url"
      value={url}
      />
      <input
      className="form-input"
      type="text"
      placeholder="Image URL"
      onChange={e => this.handleInput(e)}
      name="name"
      value={name}
      />
      <input
      className="form-input"
      type="text"
      placeholder="Image URL"
      onChange={e => this.handleInput(e)}
      name="description"
      value={description}
      />
      <input
      className="form-input"
      type="number"
      placeholder="Image URL"
      onChange={e => this.handleInput(e)}
      name="price"
      value={price}
      />
      <button
      type="submit"
      className="form-button"
      onClick={() => console.log('hi')}
      >
      Add Purchase
      </button>
    </form>
  }
}