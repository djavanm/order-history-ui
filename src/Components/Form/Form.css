form {
  margin-left: 200px;
  display: flex;
  flex-direction: column;
  width: 100%;
  justify-content: center;
}

.form-input {
  font-size: 20px;
  color: grey;
  width: 70%;
  height: 35px;
  margin: 10px;
}

button {
  width: 70%;
  margin-left: 10px;
  height: 35px;
  background: green;
  color: #ffffff;
  border: none;
}