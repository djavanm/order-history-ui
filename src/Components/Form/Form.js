import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor() {
    super()

    this.state = {
      url: '',
      name: '',
      description: '',
      price: '',
    }
  }
  
  handleInput = (e) => {
    e.preventDefault()
    this.setState({
      [e.target.name]: [e.target.value]
    })
  }

  resetForm = () => {
    this.setState({
      url: '',
      name: '',
      description: '',
      price: '',
    })
  }

  createNewPurchase = (e) => {
    e.preventDefault();
    const { url, name, description, price } = this.state;
    const newPurchase = {
      id: Date.now(),
      img: url,
      name: name, 
      description: description,
      price: price
    }
    this.resetForm();
    this.props.postNewPurchase(newPurchase);
  }

  render() {
    const {url, name, description, price} = this.state;
    return ( 
      <form className="form">
        <input
        className="form-input"
        type="text"
        placeholder="Image URL"
        onChange={e => this.handleInput(e)}
        name="url"
        value={url}
        />
        <input
        className="form-input"
        type="text"
        placeholder="Name"
        onChange={e => this.handleInput(e)}
        name="name"
        value={name}
        />
        <input
        className="form-input"
        type="text"
        placeholder="Description"
        onChange={e => this.handleInput(e)}
        name="description"
        value={description}
        />
        <input
        className="form-input"
        type="number"
        placeholder="Price"
        onChange={e => this.handleInput(e)}
        name="price"
        value={price}
        />
        <button
        type="submit"
        className="form-button"
        onClick={e => this.createNewPurchase(e)}
        >
        Add Purchase
        </button>
      </form>
    )
  }
}

export default Form;