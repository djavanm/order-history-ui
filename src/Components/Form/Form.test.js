import React from 'react';
import Form from './Form';
import { shallow } from 'enzyme';

describe('Form' , () => {
  let mockData;
  beforeEach(() => {
    mockData = {
      id: 1,
      name: 'Djavan',
      price: 9999, 
      description: 'Great for the whole family',
      img: 'random corner of the internet'
    }
  })
  it('should match the snapshot', () => {
    const wrapper = shallow(<Form postNewPurchase={jest.fn()}
    />)
    expect(wrapper).toMatchSnapshot();
  })
});