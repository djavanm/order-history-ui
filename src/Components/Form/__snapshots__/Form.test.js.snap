// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form
  className="form"
>
  <input
    className="form-input"
    name="url"
    onChange={[Function]}
    placeholder="Image URL"
    type="text"
    value=""
  />
  <input
    className="form-input"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="form-input"
    name="description"
    onChange={[Function]}
    placeholder="Description"
    type="text"
    value=""
  />
  <input
    className="form-input"
    name="price"
    onChange={[Function]}
    placeholder="Price"
    type="number"
    value=""
  />
  <button
    className="form-button"
    onClick={[Function]}
    type="submit"
  >
    Add Purchase
  </button>
</form>
`;
